x-pack/plugins/apm/public/components/shared/redirect_with_default_date_range/index.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ReactElement } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { useDateRangeRedirect } from '../../../hooks/use_date_range_redirect';
+
+export function RedirectWithDefaultDateRange({
```

Comment:
Can you add a description to this. The purpose of this function was not clear to me until I read the PR description.

Perhaps some like:
```js
// intercept invalid routes before they are rendered, and redirect, without rendering child components that expect valid & parsed routing parameters
```

Reply:
I'm also wondering what an "invalid route" is in this context. Are we specifically referring to routes that require date range and don't have it? And isn't this something io-ts can handle for us?